### PR TITLE
add pages for building, migratins, improving, debugging, evaluating a…

### PR DESCRIPTION
**Pull request by anyacherniss**
…gents

### CODE REVIEW

File: docs/content/docs/agents/improving.mdx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+2. **Use annotations** to mark which outputs are better and why (keep reading to learn more about annotations!)
+3. **Ask Claude Code** to analyze the results:
+
+```
+Analyze the results of experiment/019885bb-24ea-70f8-c41b-0cbb22cc3c00 
+and recommend which configuration performs best specifically for both accuracy and cost
+```
+
+
+# Annotations
```
Comment (anyacherniss) [inline docs/content/docs/agents/improving.mdx:106]: @pierrevalade let me know if you feel strong that annotations hsould have their own page. The placement I have for them here (in the improving agents page) is just a proposal, not a hill I care to die on :)

File: docs/content/docs/agents/migrating.mdx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import { Step, Steps } from 'fumadocs-ui/components/steps';
+import { Tabs, Tab } from 'fumadocs-ui/components/tabs';
+import { BarChart3, UploadCloud, Eye, PlugZap, ListChecks, Globe, ShieldCheck } from 'lucide-react';
+
+<Steps>
+<Step>
+
+### Update your base URL
+
+The only change you need to make is updating the `base_url`.
```
Comment (anyacherniss) [inline docs/content/docs/agents/migrating.mdx:18]: Am I missing another step?

File: docs/content/docs/agents/debugging.mdx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+This completion anotherai/completion/0198c34b-ff24-73cb-57d8-a67851e0cf10 
+input tone was enthusiastic, but the rewritten email isn't very enthusiastic. 
+Help me understand what's going wrong.
+```
+
+</Step>
+
+<Step>
+
+### Claude Does the Rest!
```
Comment (pierrevalade) [inline docs/content/docs/agents/debugging.mdx:44]: @anyacherniss we should show how to use `/model` and select Opus as well.

Comment (anyacherniss) [reply docs/content/docs/agents/debugging.mdx:44]: Addressed ✅ Text edit was in a line lower than this review comment, which is why this is not showing as outdated

File: docs/content/docs/agents/migrating.mdx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+- **Prompt Deployments**: Update prompts without code changes
+
+</Step>
+</Steps>
+
+## Next Steps
+
+- [Evaluating Agents](/agents/evaluating) - Test and compare agent performance
+- [Improving Agents](/agents/improving) - Optimize prompts and model selection
+- [Deployments](/deployments/index) - Update prompts without changing code
```
[the remark this answers is not in the document]
Comment (anyacherniss) [reply docs/content/docs/agents/migrating.mdx]: Included in the commit I pushed addressing other feedback ✅

File: docs/content/docs/agents/migrating.mdx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    base_url="https://api.anotherai.com/v1",
+)
+
+# Your existing code works as-is
+response = client.chat.completions.create(
+    model="gpt-4o",
+    messages=[
+        {"role": "user", "content": "Hello, world!"}
+    ]
+)
```
[the remark this answers is not in the document]
Comment (anyacherniss) [reply docs/content/docs/agents/migrating.mdx:59]: added! I think this comment is not showing as outdated because the edits were made on other lines

File: docs/content/docs/agents/migrating.mdx [hunk 6]
```diff
@@ -0,0 +1,94 @@
+---
+title: Migrating an Existing Agent
+summary: Guide for migrating agents from other platforms or updating existing AnotherAI agents
+description: Learn how to migrate agents from other platforms or upgrade existing agents with new features and configurations
+---
+
+## Migrating From OpenAI to AnotherAI
```
[the remark this answers is not in the document]
Comment (anyacherniss) [reply docs/content/docs/agents/migrating.mdx:7]: Good point, added!

File: docs/content/docs/agents/improving.mdx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+- What is not working (e.g. "The description of the events are too verbose, and this model missed out on extracting the updated time of the team sync")
+
+Using text-based annotations allow you to provide thorough, nuanced feedback in cases where a completion's quality isn't straightforward. For example:
+1. If you don't consider a completion as all good or all bad, you can highlight parts of a completion that are working well and parts that are not.
+2. You can add specific thoughts and context to a completion so your coding agent will have an in-depth understanding of the completion's quality.
+
+However if you would like to incorporate more quantitative ratings, you can do that by using scores, which are described below!
+
+### Scores
+Scores can be added to annotations to provide quantitative evaluations. There are no predefined scores keys, you can add whatever is important to you. Some examples of common score metrics are:
```
[the remark this answers is not in the document]
Comment (anyacherniss) [reply docs/content/docs/agents/improving.mdx:130]: ✅

File: docs/content/docs/agents/evaluating.mdx [hunk 8]
```diff
@@ -0,0 +1,414 @@
...
+    )
+```
+</Step>
+
+<Step>
+### Run the Evaluation Pipeline
+
+Evaluate your agent across multiple models:
+
+```python
```
Comment (anyacherniss) [inline docs/content/docs/agents/evaluating.mdx:113]: example code may benefit from engineer review. I copied the files in https://github.com/WorkflowAI/playground/pull/190/files and gave to Claude as a reference.

File: docs/content/docs/agents/evaluating.mdx [hunk 9]
```diff
@@ -0,0 +1,438 @@
...
+- **Current Relevance**: Always testing on the latest usage patterns
+- **No Maintenance**: No need to update test cases manually
+
+### Drawbacks
+- **Inconsistency**: Different data each time makes comparison harder
+- **No Edge Cases**: Might miss rare but important scenarios
+- **Privacy Concerns**: May need to filter sensitive production data
+- **Availability**: Requires existing completions to evaluate
+
+### Example Implementation
```
Comment (anyacherniss) [inline docs/content/docs/agents/evaluating.mdx:362]: Unsure about the options here esp. the code. Will try to test it out tomorrow. But would appreciate a second opinion in case I'm going way off in the wrong direction here.

File: docs/content/docs/agents/debugging.mdx [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+import { Step, Steps } from 'fumadocs-ui/components/steps';
+
+If you noticed issues with your agent, the easiest way to debug is to use Claude Code (or your preferred AI coding agent) to investigate the issue.
+
+<Steps>
+<Step>
+
+### Identify the Problem
+
+Find the completion(s) in AnotherAI that have the issues you want to address
```
Comment (pierrevalade) [inline docs/content/docs/agents/debugging.mdx:18]: there are situations where you don't need to find the completions.
for example, when a `user_email` is added as `metadata` on agents, you can say:
> find why the agent that does ... did not work well for customer email@... and ....

File: docs/content/docs/agents/debugging.mdx [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+</Step>
+
+<Step>
+
+### Claude Does the Rest!
+
+Claude Code will debug for you by examining the completion and agent details and input variables. After the issue is identified, you can use the Claude Code to help you rewrite your agent's code to fix the issue.
+
+<Callout type="tip">
+Based on our experience: for complex debugging tasks, we strongly recommend switching from your default model to Claude Opus using the `/models` command in Claude Code.
```
Comment (pierrevalade) [inline docs/content/docs/agents/debugging.mdx:49]:
```suggestion
Based on our experience: for complex debugging tasks, we strongly recommend switching from your default model to Claude Opus using the `/model` command in Claude Code. 
```

File: docs/content/docs/agents/evaluating.mdx (hunk 9 above)
Comment (anyacherniss) [reply docs/content/docs/agents/evaluating.mdx:362]: Update: Asked Claude to create LLM as a judge based off of the documentation and it looks like it worked. ✅

File: docs/content/docs/agents/building.mdx [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+## Getting Started
+
+Because AnotherAI is an OpenAI-compatible proxy, building an agent has minimal configuration changes from standard OpenAI implementations. However there are a few things you need to know to get started.
+
+### Base URL and API Key Setup
+
+AnotherAI provides a unified API that routes requests to various AI providers. To use AnotherAI instead of calling OpenAI directly, you need to:
+
+1. **Change the base URL** - This redirects API calls from OpenAI's servers to AnotherAI, which then routes them to the appropriate provider while adding observability features
+2. **Configure API keys** - Either your own provider keys (self-hosted) or an AnotherAI key (cloud-hosted)
```
[the remark this answers is not in the document]
Comment (anyacherniss) [reply docs/content/docs/agents/building.mdx:16]: Oh thanks for catching, I didn't realize. Follow up question: If the user is self hosting then do they need an additional API key for the specific provider? Or is there another way the agent would be run?

[the remark this answers is not in the document]
Comment (guillaq) [reply docs/content/docs/agents/building.mdx:16]: If the user is self hosting, they need to configure provider keys. But they are not configured in the code.

**Comment by pierrevalade** [comment]
@anyacherniss what's the latest? thank you.

**Comment by anyacherniss** [comment]
This was the PR I specifically mentioned in the catchup call that was waiting for Guillaume to do a final pass on. It's been waiting until he's wrapped up with higher priority todos.
